### PR TITLE
Added a quick log-out button

### PR DESCRIPTION
**Pull request by Shubhamverma2796**
Pull Request for Issue #37109.

### Summary of Changes
Added a quick log-out button.

### Testing Instructions
Try to log out using the quick log out button outside the dropdown.

### Expected result AFTER applying this Pull Request

see #37109

### CODE REVIEW

File: administrator/modules/mod_user/tmpl/default.php [hunk 1]
```diff
@@ -6,9 +6,7 @@
  * @copyright   (C) 2019 Open Source Matters, Inc. <https://www.joomla.org>
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
-
 defined('_JEXEC') or die;
-
```
Comment (richard67) [inline administrator/modules/mod_user/tmpl/default.php]: Please revert this change (removal of the 2 empty lines).

**Comment by richard67** [comment]
Shouldn't the logout then be removed from the dropdown because the entry there is redundant when this new button is implemented?

File: administrator/modules/mod_user/tmpl/default.php (hunk 1 above)
Comment (Shubhamverma2796) [reply administrator/modules/mod_user/tmpl/default.php]: done

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 65cc20b991869395629f3d14f6de550565e17dda<br /><br />Always test your changes before asking others to do so<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37183">issues.joomla.org/tracker/joomla-cms/37183</a>.</sub>

**Comment by Shubhamverma2796** [comment]
> Shouldn't the logout then be removed from the dropdown because the entry there is redundant when this new button is implemented?

I've removed the log out from the drop down.

**Comment by Shubhamverma2796** [comment]
> I have tested this item 🔴 unsuccessfully on [65cc20b](https://github.com/joomla/joomla-cms/commit/65cc20b991869395629f3d14f6de550565e17dda)Always test your changes before asking others to do so
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/37183](https://issues.joomla.org/tracker/joomla-cms/37183).

Can you please explain why?

**Comment by brianteeman** [comment]
Try to use your code and you will see

File: administrator/modules/mod_user/tmpl/default.php [hunk 2]
```diff
@@ -52,6 +52,7 @@
 			<span class="icon-universal-access icon-fw" aria-hidden="true"></span>
 			<?php echo Text::_('MOD_USER_ACCESSIBILITY_SETTINGS'); ?>
 		</a>
+		<?php $route = 'index.php?option=com_login&task=logout&amp;' . Session::getFormToken() . '=1'; ?>
```
Comment (brianteeman) [inline administrator/modules/mod_user/tmpl/default.php]: dont you think it would be better for this to be in the div where it is used instead of being in a completely different div

File: administrator/modules/mod_user/tmpl/default.php [hunk 3]
```diff
@@ -52,10 +52,11 @@
 			<span class="icon-universal-access icon-fw" aria-hidden="true"></span>
 			<?php echo Text::_('MOD_USER_ACCESSIBILITY_SETTINGS'); ?>
 		</a>
-		<?php $route = 'index.php?option=com_login&task=logout&amp;' . Session::getFormToken() . '=1'; ?>
-		<a class="dropdown-item" href="<?php echo Route::_($route); ?>">
+	</div>
+	<div class="header-item-content dropdown header-profile">
+	<?php $route = 'index.php?option=com_login&task=logout&amp;' . Session::getFormToken() . '=1'; ?>
+		<a href="<?php echo Route::_($route); ?>" title="Log Out">
```
Comment (brianteeman) [inline administrator/modules/mod_user/tmpl/default.php]: what about those people that dont speak english?

File: administrator/modules/mod_user/tmpl/default.php (hunk 2 above)
Comment (Shubhamverma2796) [reply administrator/modules/mod_user/tmpl/default.php]: Yes, I  just changed its position.

File: administrator/modules/mod_user/tmpl/default.php [hunk 4]
```diff
@@ -52,10 +52,11 @@
 			<span class="icon-universal-access icon-fw" aria-hidden="true"></span>
 			<?php echo Text::_('MOD_USER_ACCESSIBILITY_SETTINGS'); ?>
 		</a>
-		<?php $route = 'index.php?option=com_login&task=logout&amp;' . Session::getFormToken() . '=1'; ?>
-		<a class="dropdown-item" href="<?php echo Route::_($route); ?>">
+	</div>
+	<div class="header-item-content dropdown header-profile">
```
Comment (brianteeman) [inline administrator/modules/mod_user/tmpl/default.php]: are these classes correct and needed or did you just copy them from the parent div?

File: administrator/modules/mod_user/tmpl/default.php (hunk 3 above)
Comment (Shubhamverma2796) [reply administrator/modules/mod_user/tmpl/default.php]: so should I remove the title attribute?

File: administrator/modules/mod_user/tmpl/default.php (hunk 4 above)
Comment (Shubhamverma2796) [reply administrator/modules/mod_user/tmpl/default.php]: It should be header-item. Right?

File: administrator/modules/mod_user/tmpl/default.php (hunk 3 above)
Comment (brianteeman) [reply administrator/modules/mod_user/tmpl/default.php]: No then that would make it even worse. Look at the code. Look at how all language strings are used.

I am not being awkward I want you to learn by discovery not by me doing it for you.

Comment (Shubhamverma2796) [reply administrator/modules/mod_user/tmpl/default.php]: Thanks for your help. And I don't want you to do it for me. I am here to learn and contribute. I will try to fix it asap.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on d40bf363281ecbbdba090ff1405fa64442e5531f<br /><br />@chmst could you check A11Y please. Thank you.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37183">issues.joomla.org/tracker/joomla-cms/37183</a>.</sub>

**Comment by brianteeman** [comment]
@ChristineWk There are quite a few changes that need to be made

**Comment by ChristineWk** [comment]
@brianteeman 
Ah. Didn't see the changes after d40bf36. But now under github (not on issue view). Will revert test result.

**Comment by ChristineWk** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37183">issues.joomla.org/tracker/joomla-cms/37183</a>.</sub>

File: administrator/modules/mod_user/tmpl/default.php (hunk 3 above)
Comment (Shubhamverma2796) [reply administrator/modules/mod_user/tmpl/default.php]: I've changed the title from "log out" to "JLOGOUT"

**Comment by Shubhamverma2796** [comment]
@ChristineWk can you please test it again. Thanks.

**Comment by chmst** [comment]
@Shubhamverma2796 

1) The comment of @simbus82 about the icon also is valid. https://github.com/joomla/joomla-cms/issues/37109#issuecomment-1047691320 A user does not switch power on off, but he signs out. So the icon-exit would be fine for me, at least for users with ltr languages. Could you use icon-exit?
@brianteeman are we using a power-on-off icon due to rtl and ltr language, als exit has an arrow-right? 

2) It would be nice if the icon has a dark background as the other icons in the top bar.

**Comment by brianteeman** [comment]
Sorry I have no idea why that icon is being used

**Comment by pritam825** [comment]
I have tested this item :white_check_mark: successfully on 49c703be60026a6d522b760d9e152d9063867151<br /><br />Working As Expected <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37183">issues.joomla.org/tracker/joomla-cms/37183</a>.</sub>

**Comment by PhilETaylor** [comment]
The real question is does Joomla really need this? All it does is remove a single click (to open the existing menu) but yet takes up a considerable amount of pixels in an already crowded (and 3pd already using) toolbar - and it cannot be disabled.

The correct implementation would be for the User Menu to be a real menu, controlled by Joomla Menus Component, and for this button to be implemented as a module that could be enabled, positioned, and disabled at the will of the admin. 

Hard coding these changes in the way these several PR's suggest, is a backward step.

**Comment by chmst** [comment]
Fully agree 
> @PhilETaylor The correct implementation would be for the User Menu to be a real menu, controlled by Joomla Menus Component, that items can be added/removed from, and for this button to be implemented as a new module that could be enabled, positioned, and disabled at the will of the admin.

For the moment this could be a quick reminder for users to log out.

**Comment by brianteeman** [comment]
> For the moment this could be a quick reminder for users to log out.

But not an action that is really needed

**Comment by PhilETaylor** [comment]
> For the moment this could be a quick reminder for users to log out.

It is actually more secure to stay logged in. This is why GitHub and other major web applications have a long long session time and you very very rarely have to login again. The days of reminding people to logout and close their browser are history.

**Comment by Shubhamverma2796** [comment]
So should I close this pr?

**Comment by ChristineWk** [comment]
You are both right. But test itself is successful with nice button.
````<span class="icon-exit icon-fw" aria-hidden="true"></span>````<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37183">issues.joomla.org/tracker/joomla-cms/37183</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 453e34b8ed400301403a56a988bc83856edc5f56<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37183">issues.joomla.org/tracker/joomla-cms/37183</a>.</sub>

File: administrator/modules/mod_user/tmpl/default.php [hunk 5]
```diff
@@ -52,10 +52,13 @@
...
-		<?php $route = 'index.php?option=com_login&task=logout&amp;' . Session::getFormToken() . '=1'; ?>
-		<a class="dropdown-item" href="<?php echo Route::_($route); ?>">
-			<span class="icon-power-off icon-fw" aria-hidden="true"></span>
-			<?php echo Text::_('JLOGOUT'); ?>
+	</div>
+	<div class="header-item">
+	<?php $route = 'index.php?option=com_login&task=logout&amp;' . Session::getFormToken() . '=1'; ?>
+		<a href="<?php echo Route::_($route); ?>" title="<?php echo Text::_('JLOGOUT'); ?>">
+		<div class="header-item-icon">
+			<span class="icon-exit icon-fw" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/modules/mod_user/tmpl/default.php:60]:
```suggestion
			<span class="icon-exit icon-fw" aria-hidden="true"></span>
			<span class="visually-hidden"><?php echo Text::_('JLOGOUT'); ?></span>
```

**Comment by brianteeman** [comment]
@chmst did some research into the icon. It would seem that a lot of places dont find the exit door icon clear as it is rarely used. Most places just use the text. Even fontawesome themselves cant find it an obvious icon to use  as they have included text on their own ite

image

**Comment by chmst** [comment]
@brianteeman I have apps where this icon is used, and found it very clear. But I don't  insist here, I have no statistics or studies how users understand this icon. 

grafik

In general I think that a text + icon would be great, in an extra module, i really like this idea. 
But then the personal settings of the user should be removed from the top bar and go to the left menu. 

We could keep this nice idea as a feature request for further improvement?

**Comment by brianteeman** [comment]
for me personally I dont like this little button. I hit it a few times today with my fat fingers when trying to activate the user dropdown and had to log back in

**Comment by bembelimen** [comment]
Hello @Shubhamverma2796 thank you very much for your contribution. I know how much time you spend here, but after reading all the arguments and thinking about it, I have to agree that this button brings not the big benefit we hoped for. This evaluation could only have been made because of your great work, so don't feel rejected as I close this PR now. Looking forward for your next PR.

**Comment by richard67** [comment]
@bembelimen What to do with the issue #37109 ? Leave it closed without comment? Or leave some comment there? Or move it to discusssions?

**Comment by bembelimen** [comment]
As this PR is linked, isn't this enough?

**Comment by richard67** [comment]
Don't know.

**Comment by crimle** [comment]
I am deeply disappointed with the way this PR has ended. Developers should empathize with the needs of users and not exclusively think PHP and Javascript. I have been a Joomla power user for many years (back to Joomla 1.x). I currently maintain 92 Joomla websites, 30 are on Joomla 3.10 and 62 are running Joomla 4.1. 

What bothers me about Joomla 4 is the increasing number of mouseclicks that are needed for simple commands like «Log out» or «Save as copy» or «Save & New». Even the «Trash» command has been hidden behind an action button. 

I am logging in and logging out several times a day. Joomla 3 had a one-click logout button at the bottom of the screen. With Joomla 4 I need to click the user menu, then move the mouse down a couple of inches, then click again. Together with the additional mouseclicks working with articles and the many additional mouseclicks Windows 11 demands, this is awkward and is tending to cause a tendinitis to my right arm.

So I am going to do now a backend template override for my 92 websites. But from the usability point of view, it should be the objective of any modern app to provide an as comfortable and easy handling as possible.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37183">issues.joomla.org/tracker/joomla-cms/37183</a>.</sub>

**Comment by Shubhamverma2796** [comment]
> 

First of all, it's not at all discouraging. I'm here to learn and contribute, so closing a pr is not a big deal. And, I totally agree with the https://github.com/joomla/joomla-cms/pull/37183#issuecomment-1061236118 because I encountered the same issue today, got logged out by mistake because of the button. But still, I believe that the log-out button should be implemented in such a way that it is clearly and directly visible to the end-user, instead of forcing the user to click unnecessarily.

**Comment by pritam825** [comment]
@Shubhamverma2796 can't we do like, when we will click on the logout button it will show the alert , are you sure want to logout, like the Facebook used for logout, so that user will see easily logout button and because of alert it won't create the problem like the brainteeman faced

**Comment by Shubhamverma2796** [comment]
> @Shubhamverma2796 can't we do like, when we will click on the logout button it will show the alert , are you sure want to logout, like the Facebook used for logout, so that user will see easily logout button and because of alert it won't create the problem like the brainteeman faced

sounds good to me. But still the number of click remains the same 😅

**Comment by richard67** [comment]
So there would be again 2 clicks necessary like without any of these PRs. So where would be the advantage then?

**Comment by pritam825** [comment]
User can easily see the logout button

**Comment by pritam825** [comment]
The advantage would be that the logout button is visible directly. When I first used joomla it took me few minutes to find the logout button

**Comment by brianteeman** [comment]
There is no need to log out
